Tensile/Source/lib/include/Tensile/PlaceholderLibrary.hpp

```diff
@@ -0,0 +1,169 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright 2022 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <Tensile/MasterSolutionLibrary.hpp>
+#include <Tensile/SolutionLibrary.hpp>
+#include <Tensile/Tensile.hpp>
+
+#include <algorithm>
+
+namespace Tensile
+{
+    // Which placeholder libraries should be initialized at start
+    // To be extended in the future
+    enum class LazyLoadingInit
+    {
+        All
+    };
+
+    //Regex patterns for initializing libraries on startup
+    inline std::string RegexPattern(LazyLoadingInit condition)
+    {
+        switch(condition)
+        {
+        case LazyLoadingInit::All:
+            return "TensileLibrary_.*";
+        }
+
+        return "";
+    }
+
+    template <typename MyProblem, typename MySolution = typename MyProblem::Solution>
+    struct PlaceholderLibrary : public SolutionLibrary<MyProblem, MySolution>
+    {
+        mutable std::shared_ptr<SolutionLibrary<MyProblem, MySolution>> library;
+        mutable SolutionMap<MySolution>*                                solutions;
+        mutable std::mutex*                                             solutionsGuard;
+        mutable std::mutex                                              lazyLoadingGuard;
+        std::string                                                     filePrefix;
+        std::string                                                     suffix;
+        std::string                                                     libraryDirectory;
+
+        PlaceholderLibrary() = default;
+
+        bool loadPlaceholderLibrary() const
+        {
+            std::lock_guard<std::mutex> lock(lazyLoadingGuard);
```

Comment:
For example maybe the ones that aren't a pointer could be called m_lazyLoadingMutex .